src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -0,0 +1,102 @@
+import io.iohk.ethereum.db.storage.NodeStorage
+import io.iohk.ethereum.domain.{Block, BlockHeader, Receipt, SignedTransaction, _}
+import io.iohk.ethereum.ledger.Ledger.{BlockPreparationResult, BlockResult}
+import io.iohk.ethereum.ledger.{BlockPreparationError, BloomFilter, Ledger}
+import io.iohk.ethereum.mining.BlockGenerator.{InvalidOmmers, NoParent}
+import io.iohk.ethereum.mpt.{MerklePatriciaTrie, RLPByteArraySerializable}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions._
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockHeaderImplicits.headerRlpEncDec
+import io.iohk.ethereum.network.p2p.messages.PV63.ReceiptImplicits.receiptRlpEncDec
+import io.iohk.ethereum.rlp.RLPImplicitConversions.toRlpList
+import io.iohk.ethereum.rlp.encode
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.ByteUtils.or
+import io.iohk.ethereum.validators.MptListValidator.intByteArraySerializable
+import io.iohk.ethereum.validators.OmmersValidator.OmmersError
+import io.iohk.ethereum.validators.Validators
+import org.spongycastle.crypto.AsymmetricCipherKeyPair
+import org.spongycastle.crypto.params.{ECPrivateKeyParameters, ECPublicKeyParameters}
+import io.iohk.ethereum.crypto._
+
+class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: BlockchainConfig, ledger: Ledger, validators: Validators) {
+  val difficulty = new DifficultyCalculator(blockchainConfig)
+
+  //todo add logic
+  private val fakeAddress = 42
+  private lazy val block = generateBlockForMining(1, Nil, Nil, Address(fakeAddress)).fold(e => null, identity)
+
+  def generateBlockForMining(): Block = block
+
+  def generateBlockForMining(blockNumber: BigInt, transactions: Seq[SignedTransaction], ommers: Seq[BlockHeader], beneficiary: Address):
+  Either[BlockPreparationError, Block] = {
+    val blockchain = BlockchainImpl(blockchainStorages)
+
+    validators.ommersValidator.validate(blockNumber, ommers, blockchain).left.map(InvalidOmmers).flatMap { _ =>
+      val blockTimestamp = Instant.now.getEpochSecond
+      blockchain.getBlockByNumber(blockNumber - 1).map { parent =>
+        val header = BlockHeader(
+          parentHash = parent.header.hash,
+          ommersHash = ByteString(kec256(encode(toRlpList(ommers)))),
+          beneficiary = beneficiary.bytes,
+          stateRoot = ByteString.empty,
+          transactionsRoot = buildMpt(transactions, new RLPByteArraySerializable[SignedTransaction]),
+          receiptsRoot = ByteString.empty,
+          logsBloom = ByteString.empty,
+          difficulty = difficulty.calculateDifficulty(blockNumber, blockTimestamp, parent.header),
+          number = blockNumber,
+          gasLimit = calculateGasLimit(parent.header.gasLimit),
+          gasUsed = 0,
+          unixTimestamp = blockTimestamp,
+          extraData = ByteString("mined with etc scala"),
```

Comment:
you can say the same about parity which uses ethminer and does not have own implementation ;)